src/appleseed/renderer/kernel/lighting/lighttree.cpp

```diff
@@ -281,10 +297,17 @@ float LightTree::compute_node_probability(
     // Calculate probability of a single node based on its distance
     // to the surface point being illuminated.
     // For leaf nodes use the actual position of the light source, instead
-    // of center of the bbox. It is more correct for shaped lights.
-    const Vector3d position = node.is_leaf()
-        ? m_light_sources[m_items[node.get_item_index()].m_light_source_index]->get_position()
-        : bbox.center();
+    // of center of the bbox. It is more precise for shaped lights.
+    Vector3d position;
+    if (node.is_leaf() && m_items[node.get_item_index()].m_light_type == EmittingTriangleType)
```

Comment:
Factorize the item lookup (`m_items[node.get_item_index()]`) to avoid doing it twice.